src/components/MAPI/permissions/utils.ts

```diff
@@ -301,3 +302,66 @@ export function hasAppAccess(
     hasAppAccesInNamespace(permissions, ns)
   );
 }
+
+/**
+ * Fetch access to a resource within a given namespace.
+ * Returns an object mapping each given verb to a boolean indicating access.
+ * @param httpClientFactory
+ * @param auth
+ * @param namespace
+ * @param group
+ * @param resource
+ */
+export async function fetchAccessForResource(
+  httpClientFactory: HttpClientFactory,
+  auth: IOAuth2Provider,
+  namespace: string,
+  verbs: readonly AccessControlRoleItemVerb[],
+  group: string,
+  resource: string
+): Promise<Record<AccessControlRoleItemVerb, boolean>> {
+  const requests = verbs.map(async (verb) => {
+    try {
+      const request: authorizationv1.ISelfSubjectAccessReviewSpec = {
+        resourceAttributes: {
+          namespace,
+          verb,
+          group,
+          resource,
+        },
+      };
+
+      const accessReviewResponse =
+        await authorizationv1.createSelfSubjectAccessReview(
+          httpClientFactory(),
+          auth,
+          request
```

Comment:
The permissions map we use to calculate user permissions does not include the WC namespace, so we have to calculate permissions for `apps` a little differently: we create a `SelfSubjectAccessReview` in the WC namespace for a specific resource